engine/univMinim.ml

```diff
@@ -271,17 +271,23 @@ let is_bound l lbound = match lbound with
 | UGraph.Bound.Prop -> Level.is_prop l
 | UGraph.Bound.Set -> Level.is_set l
 
+(* if [is_minimal u] then constraints [u <= v] may be dropped and get
+   used only for set_minimization. *)
+let is_minimal ~lbound u =
+  Level.is_sprop u || Level.is_prop u || is_bound u lbound
+
 (* TODO check is_small/sprop *)
 let normalize_context_set ~lbound g ctx us algs weak =
   let (ctx, csts) = ContextSet.levels ctx, ContextSet.constraints ctx in
   (* Keep the Prop/Set <= i constraints separate for minimization *)
   let smallles, csts =
-    Constraint.partition (fun (l,d,r) -> d == Le && (is_bound l lbound || Level.is_sprop l)) csts
+    Constraint.partition (fun (l,d,r) -> d == Le && is_minimal ~lbound l) csts
   in
   let smallles = if get_set_minimization ()
     then Constraint.filter (fun (l,d,r) -> LMap.mem r us && not (Level.is_sprop l)) smallles
     else Constraint.empty
   in
+  let smallles = Constraint.map (fun (_,_,r) -> Level.set, Le, r) smallles in
```

Comment:
Could this be fused with the `Constraint.filter` call just above it? I am asking because I saw the slowdown in coq-engine-bench and I was wondering if maybe `smallles` is now bigger than it used to be and if that could make the double traversal of it more noticeable than before.